test/executor.test.js

```diff
@@ -373,6 +373,22 @@ describe('executor', function() {
     });
   });
 
+  describe('with boot script returning a promise and calling callback',
+    function() {
+      before(function() {
+        process.promiseAndCallback = true;
+      });
+
+      after(function() {
+        delete process.promiseAndCallback;
+      });
+
+      it('should only call the callback once', function(done) {
+        boot.execute(app, simpleAppInstructions(), done);
```

Comment:
Perhaps add a comment explaining that mocha fails the test when the callback was called multiple times?